modules/apps/portal-vulcan/portal-vulcan-test/src/testIntegration/java/com/liferay/portal/vulcan/internal/graphql/servlet/test/GraphQLServletTest.java

```diff
@@ -174,6 +176,121 @@ public void testQueryDepthLimit() throws Exception {
+			10, () -> _test(1, 10, null, 30));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			30, () -> _test(1, 20, null, null));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			30, () -> _test(1, 15, null, 15));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			30, () -> _test(1, 30, null, 30));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			30, () -> _test(1, 30, null, 40));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			50, () -> _test(2, 20, 2, null));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			50, () -> _test(3, 20, 3, null));
+
+		// Limited page size configured and unlimited page size requested
+
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			50, () -> _test(1, 50, null, -1));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			50, () -> _test(1, 50, null, 0));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			50, () -> _test(1, 50, -1, null));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			50, () -> _test(1, 50, 0, null));
+
+		// Unlimited page size configured and limited page size requested
+
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			-1, () -> _test(1, 20, null, null));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			-1, () -> _test(1, 25, null, 25));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			-1, () -> _test(2, 20, 2, null));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			-1, () -> _test(2, 25, 2, 25));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			0, () -> _test(1, 20, null, null));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			0, () -> _test(1, 25, null, 25));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			0, () -> _test(2, 20, 2, null));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			0, () -> _test(2, 25, 2, 25));
+
+		// Unlimited page size configured and unlimited page size requested
+
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			-1, () -> _test(-1, -1, -1, null));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			-1, () -> _test(-1, -1, 0, null));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			-1, () -> _test(-1, -1, null, -1));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			-1, () -> _test(-1, -1, null, 0));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			0, () -> _test(-1, -1, -1, null));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			0, () -> _test(-1, -1, 0, null));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			0, () -> _test(-1, -1, null, -1));
+		PaginationConfigurationTestUtil.withPageSizeLimit(
+			0, () -> _test(-1, -1, null, 0));
+	}
+
+	private void _test(
+			int expectedPage, int expectedPageSize, Integer requestPage,
+			Integer requestPageSize)
+		throws Exception {
+
+		JSONObject jsonObject = JSONUtil.getValueAsJSONObject(
+			invoke(
+				new GraphQLField(
+					"testDTOPage",
+					HashMapBuilder.put(
+						"page", (Object)requestPage
```

Comment:
Do you need to cast here to Object? I think by setting the type <String, Integer> would be enough